samples/bluetooth/ipsp/README.rst

```diff
@@ -27,14 +27,132 @@ following commands:
+             inet6 addr: fe80::fa2f:a8ff:feeb:6d8c/64 Scope:Link
+             UP POINTOPOINT RUNNING MULTICAST  MTU:1280  Metric:1
+             RX packets:2 errors:0 dropped:3 overruns:0 frame:0
+             TX packets:6 errors:0 dropped:0 overruns:0 carrier:0
+             collisions:0 txqueuelen:1000
+             RX bytes:92 (92.0 B)  TX bytes:233 (233.0 B)
+
+As can be seen from the output, only a link-local IPv6 address was assigned
+to the interface.
+
+At this point, you can already test IPv6 connectivity (and discover your board's IPv6
+address) by pinging "All local-link nodes" IPv6 address:
+
+.. code-block:: console
+
+   # ping6 -I bt0 ff02::1
+   PING ff02::1(ff02::1) from fe80::fa54:a8ff:feeb:218f bt0: 56 data bytes
+   64 bytes from fe80::fa54:a8ff:feeb:218f: icmp_seq=1 ttl=64 time=0.088 ms
+   64 bytes from fe80::c9af:14ff:fe57:d86e: icmp_seq=1 ttl=64 time=285 ms (DUP!)
+
+For each ping packet, both your host and BLE board send reply, so you can see
+board's reply marked as "DUP!". We can now ping the board directly:
+
+.. code-block:: console
+
+   # ping6 fe80::c9af:14ff:fe57:d86e%bt0
+   PING fe80::c9af:14ff:fe57:d86e%bt0(fe80::c9af:14ff:fe57:d86e) 56 data bytes
+   64 bytes from fe80::c9af:14ff:fe57:d86e: icmp_seq=1 ttl=64 time=177 ms
+   64 bytes from fe80::c9af:14ff:fe57:d86e: icmp_seq=2 ttl=64 time=53.0 ms
+
+Note that command above uses "scoped IPv6 address", where scope is defined by
+networking interface, with ``%bt0`` appended in this case. Specification like
+that is in alternative to passing ``-I bt0`` to ``ping6`` (and works with
+other networking tools like ``telnet``, ``nc``, ``curl``, etc.)
+
+While we can use link-local address, it's not very convenient, as it must be
+scoped and changes on each run. Instead, IPSP sample is configured with
+``2001:db8::1`` static address, so let's configure host's interface to let
+access that address (for that, we'll need to configure bt0 with address
+``2001:db8::2``):
+
+.. code-block:: console
+
+   # ip address add 2001:db8::2/64 dev bt0
+
+We now should be able to ping static address:
+
+.. code-block:: console
+
+   # ping6 2001:db8::1
+   PING 2001:db8::1(2001:db8::1) 56 data bytes
+   64 bytes from 2001:db8::1: icmp_seq=1 ttl=64 time=282 ms
+
+The IPSP sample includes builtin echo server for UDP and TCP on a port 4242,
+which we can test with:
+
+.. code-block:: console
+
+   $ telnet 2001:db8::1 4242
+   Trying 2001:db8::1...
+   Connected to 2001:db8::1.
+   Escape character is '^]'.
+   test
+   test
+   test2
+   test2
+   ^]
+   
+   telnet> quit
+   Connection closed.
+
+In the output above, first ``test`` line was typed, next was echoed back by
+the board. Likewise for ``test2``. To quit telnet tool, type Ctrl+], then
+"quit" at the prompt.
+
+As an alternative to using well-known networking tools above, and also to
+test both TCP and UDP echo, you can use Zephyr's helper tool for ``net-tools``
```

Comment:
how about:
    
     Zephyr's helper tool in the GitHub ``zephyrproject-rtos/net-tools`` repository: